cmd/zoekt-sourcegraph-indexserver/queue.go

```diff
@@ -2,28 +2,33 @@ package main
 
 import (
 	"container/heap"
+	"encoding/gob"
+	"fmt"
+	"io"
+	"net/http"
 	"reflect"
+	"sort"
 	"sync"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promauto"
 )
 
 type queueItem struct {
-	// repoID is the ID of the repo
-	repoID uint32
-	// opts are the options to use when indexing repoID.
-	opts IndexOptions
+	// RepoID is the ID of the repo
```

Comment:
These fields needed to be exported in order for gob to be able to properly read them.

Reply:
https://pkg.go.dev/encoding/gob

> ...Structs encode and decode only exported fields...

Reply:
Do we need to export all them? EG We don't seem to send `Indexed` to the client.

Reply:
I didn't see a good reason to leave one of these fields unexported since the type itself is unexported. Even though we don't use it in the CLI output for now, I can see us adding more columns in the CLI output, or returning more of this information in the html/json version of this output.